### ISSUE TITLE
Release Android APK or publish to Fdroid repo

### ISSUE THREAD

**Issue by krillin666** [omnivore-app/omnivore#1853]
Hi !

First, thank you so much for the work being done here.

I know the Android app is still lagging a bit behind the iOS one, but this being an open source project it would be essential for users to be able to install this outside Google Play.

For that you could release the app to an Fdroid repo, or instead release the compiled APK in this repo so users could use Obtainium (https://github.com/ImranR98/Obtainium) to automatically update the APK on new releases.

Keep up the amazing work !

**Comment by jacksonh** [comment]
Thanks, adding some notes here on F-Droid: https://f-droid.org/en/docs/Submitting_to_F-Droid_Quick_Start_Guide/ 

One challenge is the social login is tied to our app signing, we might need to create new social login profiles for this version

**Comment by linsui** [comment]
> One challenge is the social login is tied to our app signing

What does this mean?

GMS is used so this can't be published to F-Droid. You may need to create a FOSS flavor.

**Comment by jacksonh** [comment]
> > One challenge is the social login is tied to our app signing
> 
> What does this mean?
> 
> GMS is used so this can't be published to F-Droid. You may need to create a FOSS flavor.

Thanks. I suspect this is just for Google login, which we I think can remove on build. Maybe for notifications also.

**Comment by krillin666** [comment]
> > > One challenge is the social login is tied to our app signing
> > 
> > 
> > What does this mean?
> > GMS is used so this can't be published to F-Droid. You may need to create a FOSS flavor.
> 
> Thanks. I suspect this is just for Google login, which we I think can remove on build. Maybe for notifications also.

Having Google dependencies stripped would be the best, so it can reach much more users through Fdroid. But in the meantime you could publish the compiled APK, so people could use Obtainium to get updates.

**Comment by satindar** [comment]
Thanks for that comment @krillin666 

Looking into Obtanium right now. I think this might be a better option for us since a few other people asked for the APK.

**Comment by satindar** [comment]
Ok I think as the app developer we just need to add the APK to our Github releases page. Nothing specific required from us for a user to access that via Obtanium (or other install methods that use Github release pages).

**Comment by satindar** [comment]
and I just now noticed you mentioned Obtanium when you first opened this issue...sorry for not reading carefully! 🤦🏽

**Comment by daviddavo** [comment]
Another option to make the non-foss flavor available on f-droid is using [IzzyOnDroid](https://apt.izzysoft.de/fdroid/index/info)

There is a lot of people that have this repository added to f-droid to download apps like logseq

**Comment by jacksonh** [comment]
Hello, i started some of the work to publish to F-droid last night, I think I should have the PR submitted by the weekend.

**Comment by xplosionmind** [comment]
> I think I should have the PR submitted by the weekend.

I fear this issue got lost in the meantime, and there is no PR open at the moment…

**Comment by mangkoran** [comment]
Any update regarding this ticket? Having the APK in the repo allow many users to directly install the app. F-Droid is good, but it has its own requirements that need to be fulfilled before an app can be published there.

**Comment by jacksonh** [comment]
Yeah, unfortunately we can't post to F-Droid because of their requirements and some libraries we use. We could post the APK here, but people wouldn't be able to use the social logins.

**Comment by linsui** [comment]
Many F-Droid users don't use Google. That's why they use F-Droid - to de-google. And the IzzySoft repo has less requirement. You just need to upload your apk to the release and they can pick up it.

**Comment by jacksonh** [comment]
@linsui and do you need the apk to be unsigned?

**Comment by jacksonh** [comment]
OK first pass here: https://github.com/omnivore-app/omnivore/releases/tag/android-0.0.101

**Comment by linsui** [comment]
Your apk is too large for Izzy's repo. Not sure if it will be small enough with only one abi.

**Comment by IzzySoft** [comment]
Way to large, and unlikely to get into the allotted size even for per-ABI builds, sorry. My estimate is each APK might get down to 70 MB, with much luck, which is still more than twice the per-app size allowed in my repo. Let's have a look:

* native stuff is about 20 MB per ABI, so per-ABI-builds would be around 90..100 MB
* assets make up 40 MB (uncompressed)
* DEX take another 40 MB (uncompressed)
* I see the app uses Firebase, Play Services etc – not a good start for a FOSS app. FOSS folks don't want to give their collections to Google  and as those libraries are proprietary, you cannot even tell what they're really doing. rust what Google says? Nope. They asked for that too often and had us fooled (remember "location collection" which was not supposed to be there, for example).

TL;DR: Sorry, but no chance as far as I can see.

**Comment by Podginator** [comment]
Closing due to age, and the self-hosted nature of Omnivore. Done as a batch action. If this issue still exists, please reopen it.
